src/vnsw/opencontrail-vrouter-netns/opencontrail_vrouter_netns/vrouter_hyperv.py

```diff
@@ -0,0 +1,469 @@
+                                #self.nic_left['uuid'])[:self.NAME_LEN]
+        self.nic_right['name'] = (self.RIGHT_DEV_PREFIX + "2")
+                                #self.nic_right['uuid'])[:self.NAME_LEN]
+
+
+    def spawn_vm(self):
+        """calls powershell to spawn vm """
+        if self._vm_exists():
+            raise ValueError("Specified Windows gateway VM already exists")
+
+        self._configure_host_mgmt_ip()
+
+        powershell(["New-VM", "-Name", self.wingw_name, \
+                    "-Path", self.vm_location, \
+                    "-Generation", self.HYPERV_GENERATION, \
+                    "-MemoryStartupBytes", self.RAM_GB, \
+                    "-VHDPath", self.vhd_path, \
+                    "-SwitchName", self.mgmt_vswitch_name])
+
+        powershell(["Set-VMFirmware", "-VMName", self.wingw_name, \
+                    "-EnableSecureBoot", "Off"])
+
+        powershell(["Start-VM", "-Name", self.wingw_name])
+
+        self.new_mgmt_ip = self._generate_new_mgmt_ip()
+        self._configure_vm_mgmt_ip()
+
+        for _ in range(2):
+            powershell(["Add-VMNetworkAdapter", "-VMName", self.wingw_name,
+                        "-SwitchName", self.vrouter_vswitch_name])
+
+
+    def destroy_vm(self):
+        """calls powershell to destroy vm """
+        if not self._vm_exists():
+            raise ValueError("Specified Windows gateway VM does not exist")
+        powershell(["Stop-VM", "-Name", self.wingw_name, "-Force"])
+        powershell(["Remove-VM", "-Name", self.wingw_name, "-Force"])
+
+
+    def set_snat(self):
+        """sshs into gateway machine and configures SNAT"""
+        self.ssh_client = None
+        try:
+            self.ssh_client = paramiko.SSHClient()
+            self.ssh_client.set_missing_host_key_policy(
+                paramiko.AutoAddPolicy())
+            self.ssh_client.connect(str(self.new_mgmt_ip), 
+                                    username=self.USERNAME,
+                                    password=self.PASSWORD)
+            self._setup_snat()
+        finally:
+            if self.ssh_client:
+                self.ssh_client.close()
+
+
+    def register_to_agent(self):
+        """registers wingw interface (as seen on host) to agent"""
+        # get 2 snat ifaces of wingw VM
+        # _add_port_to_agent on those nics
+        # TODO do we have to register mgmt interface?
+        pass
+
+
+    def unregister_from_agent(self):
+        """unregisters wingw interfaces (as seen on host) from agent"""
+        self._delete_port_from_agent(self.nic_left)
+        self._delete_port_from_agent(self.nic_right)
+        # TODO do we have to unregister mgmt interface?
+        pass
+
+
+    def _vm_exists(self):
+        """calls powershell to check whether vm exists"""
+        try:
+            powershell(["Get-VM", "-Name", self.wingw_name])
+        except subprocess.CalledProcessError as e:
+            if "unable to find" in e.output:
```

Comment:
I suggest using `e.returncode` instead of parsing the string. Parsing the string isn't stable, while return codes should be consistent.